xds/internal/xdsclient/filter_chain.go

```diff
@@ -637,8 +637,7 @@ func processNetworkFilters(filters []*v3listenerpb.Filter) (*FilterChain, error)
 					}
 					filterChain.InlineRouteConfig = &routeU
 				case nil:
-					// No-op, as no route specifier is a valid configuration on
-					// the server side.
+					return nil, fmt.Errorf("no RouteSpecifier: %+v", hcm)
```

Comment:
Was this specified anywhere? In the gRFC or somewhere else?
I'm wondering why we thought it's OK to be nil.

Reply:
And it would be OK to delete this `case`? The default below would then handle nil.

Reply:
Actually, we should probably keep that. The client side returns a special error for nil https://github.com/grpc/grpc-go/blob/670c133e568e89c901d05b898af9f5ce72a88c6d/xds/internal/xdsclient/xds.go#L153

Change the comment of these two fields to `exactly one of RouteConfigName and InlineRouteConfig is set`?

Reply:
Since the processing of the Route Configuration was added due to RBAC A36 has language on validations once Route Configuration is processed server side. However, there was no explicit language on a nil RouteSpecifier/a missing Route Configuration (usually when RDS fails, which after talking with Eric we have the right behavior which is to ACK resource, but RPC's fail (no calling goodupdate.Fire()). However, all my RBAC code didn't handle the case of a nil RouteSpecifier, which lead to issue #4924. Luckily, Eric pointed to the fact that the RouteSpecifier is in fact required in the Envoy proto: https://github.com/envoyproxy/envoy/blob/56e8c45b1b340c4a4f8f02ec2488354c31806d59/api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#L317, so this fix fixes the bug in RBAC implementation as well. Changed comment.

Reply:
One thing I mentioned in the issue thread I think is important to bring up, is that I think that NACKing a missing RouteSpecifier is fair for users since the addition of RBAC added the addition of a valid route configuration (matching vh + route) for RPC's to proceed normally, and this will communicate to them the delta they need to make in their control plane resources. Especially if we enable RBAC by default.